File: youlexuan-manager-web/src/main/java/com/wd/sellergoods/controller/BrandController.java
package com.wd.sellergoods.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.wd.entity.PageResult;
import com.wd.entity.Result;
import com.wd.pojo.TbBrand;
import com.wd.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }
    @RequestMapping("/findPage")
    public PageResult findPage(int pageNum,int pageSize){
        return brandService.findPage(pageNum, pageSize);
    }
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){
        try {
            brandService.add(tbBrand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(long id){
       return brandService.findOne(id);
    }

    @RequestMapping("/del")
    public Result del(long[] ids){
        try {
            brandService.del(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand,int pageNum,int pageSize){
        return brandService.findPage(tbBrand,pageNum,pageSize);
    }
}

File: youlexuan-manager-web/src/main/webapp/js/controller/brandController.js
//创建controller
app.controller('brandController',function ($scope,$controller,brandService) {
    //继承
    $controller('baseController',{$scope:$scope});

    //查全部
    $scope.findAll=function () {
        //$http.get('../brand/findAll.do')
        brandService.findAll().success(function (resp) {
            $scope.list=resp;
        });
    }

    //保存
    $scope.save=function () {
        if($scope.entity.id==null){//id为空就是添加
            //$http.post('../brand/add.do',$scope.entity)
            brandService.add($scope.entity).success(function (resp) {
                if(resp.success){
                    //重查
                    $scope.reloadList();
                }else {
                    alert(resp.message);
                }
            });
        }else{//id不为空就是修改
            //$http.post('../brand/update.do',$scope.entity)
            brandService.update($scope.entity).success(function (resp) {
                if(resp.success){
                    $scope.reloadList();
                    $scope.entity={};
                }else{
                    alert(resp.message);
                }
            })
        }
    }

    //查询实体
    $scope.findOne=function (id) {
        //$http.get('../brand/findOne.do?id='+id)
        brandService.findOne(id).success(function (resp) {
            $scope.entity=resp;
        })
    }

    //删除
    $scope.del=function () {
        //$http.get('../brand/del.do?ids='+$scope.selectIds)
        brandService.del($scope.selectIds).success(function (resp) {
            if(resp.success){
                $scope.reloadList();
                $scope.selectIds=[];
            }else{
                alert(resp.message);
            }
        })
    }

    //条件查询
    $scope.searchEntity={};//定义一个搜索对象
    $scope.search=function (pageNum,pageSize) {
        //$http.post('../brand/search.do?pageNum='+pageNum+'&pageSize='+pageSize,$scope.searchEntity)
        brandService.search(pageNum,pageSize,$scope.searchEntity).success(function (resp) {
            $scope.paginationConf.totalItems=resp.total;//总的记录数赋值给分页
            $scope.brandList=resp.rows;//给变量赋值，数据集合
        });
    }

})